php_pgsql-freebsd-setup/azuredeploy.json

```diff
@@ -0,0 +1,1297 @@
+      }
+    },
+    "numberOfDataDisks": {
+      "type": "int",
+      "defaultValue": 16,
+      "maxValue": 64,
+      "metadata": {
+        "description": "Number of attached data disk"
+      }
+    },
+    "sizeOfDataDisksInGB": {
+      "type": "int",
+      "defaultValue": 30,
+      "maxValue": 1023,
+      "metadata": {
+        "description": "Size of attached data disk"
+      }
+    },
+    "diskCaching": {
+      "type": "string",
+      "defaultValue": "ReadWrite",
+      "allowedValues": [
+        "None",
+        "ReadOnly",
+        "ReadWrite"
+      ],
+      "metadata": {
+        "description": "Caching type of data disk"
+      }
+    }
+  },
+  "variables": {
+    "storageAccountName": "[concat(uniquestring(parameters('vmNamePrefix')), 'storage1')]",
+    "postgresqlstorageAccountName": "[concat(uniquestring(parameters('vmNamePrefix')), 'storage2')]",
+    "dataDisk1VhdName": "[concat(uniquestring(parameters('vmNamePrefix')), 'datadisk1')]",
+    "OSDiskName": "[concat(uniquestring(parameters('vmNamePrefix')), 'osdisk')]",
+    "vm1Name": "[concat(parameters('vmNamePrefix'), '-frontend')]",
+    "vm2Name": "[concat(parameters('vmNamePrefix'), '-php1')]",
+    "vm3Name": "[concat(parameters('vmNamePrefix'), '-php2')]",
+    "vm4Name": "[concat(parameters('vmNamePrefix'), '-postgresql')]",
+    "imagePublisher": "MicrosoftOSTC",
+    "imageOffer": "FreeBSD",
+    "nic1Name": "[concat(variables('vm1Name'), 'nic1')]",
+    "nic2Name": "[concat(variables('vm1Name'), 'nic2')]",
+    "nic3Name": "[concat(variables('vm2Name'), 'nic1')]",
+    "nic4Name": "[concat(variables('vm2Name'), 'nic2')]",
+    "nic5Name": "[concat(variables('vm3Name'), 'nic1')]",
+    "nic6Name": "[concat(variables('vm3Name'), 'nic2')]",
+    "nic7Name": "[concat(variables('vm4Name'), 'nic1')]",
+    "addressPrefix": "10.0.0.0/16",
+    "subnet1Name": "Subnet-1",
+    "subnet1Prefix": "10.0.0.0/24",
+    "subnet2Name": "Subnet-2",
+    "subnet2Prefix": "10.0.1.0/24",
+    "subnet3Name": "Subnet-3",
+    "subnet3Prefix": "10.0.2.0/24",
+    "publicIPAddressName": "[concat(uniquestring(parameters('vmNamePrefix')), 'publicip')]",
+    "publicIPAddressType": "Dynamic",
+    "vmStorageAccountContainerName": "vhds",
+    "virtualNetworkName": "[concat(parameters('vmNamePrefix'),'-vnet')]",
+    "vnetID": "[resourceId('Microsoft.Network/virtualNetworks',variables('virtualNetworkName'))]",
+    "subnet1Ref": "[concat(variables('vnetID'),'/subnets/',variables('subnet1Name'))]",
+    "subnet2Ref": "[concat(variables('vnetID'),'/subnets/',variables('subnet2Name'))]",
+    "subnet3Ref": "[concat(variables('vnetID'),'/subnets/',variables('subnet3Name'))]",
+    "blobUriAzureCloud": ".blob.core.windows.net",
+    "blobUriAzureChinaCloud": ".blob.core.chinacloudapi.cn",
+    "blobUri": "[variables(concat('blobUri', parameters('environment')))]",
+    "diskArray": [

```

Comment:
using managed disk for the data disks could simplify this a bit since you would no longer need to specify the vhd uri; also, I'm not sure how performant putting 64 data disks in the same storage account is.